### DISCUSSION TITLE
Reducing CPU load on Zero W v1?

### DISCUSSION THREAD

**Discussion by mntbighker** [bluenviron/mediamtx#3311] [Questions]
### Question

I have two R Pi Zero W's running Bookworm with the current MediaMTX. I'm streaming PiCam V3 to my Synology Surveillance Station. With the "standard" boot parameters, so running with boost (1Ghz I presume) they both sh|t the bed within a few minutes. If I drop the CPU's to 800 MHz they will run stable, but they run hot. So my question is, are there any obvious things to do to reduce the CPU load? I guess the Zero is marginal for this application. I don't care if I lose frame rate, but prefer to keep the full res of the two V3 cams I just bought to replace the V2's. The V2's BTW also ran way too hot. I built MTX from source, so not running in any Docker container or anything like that. If libcamera is a dependency, does it matter to performance what version of the dev libcamera framework was installed when MTX was built? Is the performance efficiency likely to be improving over time, or are things pretty optimized already? How much improvement would be seen with a Pi Zero 2W over a Zero W (v1)?

**Comment by dragulesq** [comment]
What are your video settings? (bitrate/fps/resolution) Do you have a "proper" power supply?
Here are mine:

``` 
  rpiCameraWidth: 1920
  rpiCameraHeight: 1080
  rpiCameraFPS: 24
  # period between IDR frames
  rpiCameraIDRPeriod: 72
  # bitrate
  rpiCameraBitrate: 6500000
  # H264 profile
  rpiCameraProfile: high
  # H264 level 
  rpiCameraLevel: '4.2'
```

I had some issues where it would reboot after 2-3 minutes when using newer versions of mediamtx. 1.6.0 is stable for me. I use the Adafruit power supply, a USB ethernet adapter (wifi performance was bad for me), v3 cam and using mediamtx release binaries 

```up 25 days, 23:20,  1 user,  load average: 1.98, 2.08, 2.19```


No issues with V2s and mediamtx 1.8.0

**Reply by mntbighker** [reply]
I duplicated your settings. The CPU hovers around 90% and the temp is just under 160F on both Pi's. The picture is fantastic of course. Apparently the RAM piggybacks over the CPU, so adding cooling is not necessarily effective? Besides, my 3D printed enclosures do not have much room for more heat sync. I am running 1.8.0 and CPU freq of 800. I had tried 600 but it seemed to make no temp difference, but caused other issues.

**Comment by dragulesq** [comment]
I purchased my PI Zero W1s  7 years ago along with cam v2s. I've used them initially with motion, then v4l2-rtspserver, and now with cam v3s and mediamtx. Motion had especially high CPU usage, so they probably ran hot all the time. The hardware's been rock solid :muscle: Amazing what you can get for $15!

**Comment by mntbighker** [comment]
I ordered some copper heat sinks on Amazon and these dropped the temps from 160 to 140F. 

I also ordered two Pi 2W's to add to the stable.

[a comment by an automated account is not shown]
